### ISSUE TITLE
Block inserter: Focus search input when opening the inserter, and improve its cohesion

### ISSUE THREAD

**Issue by juanfra** [WordPress/gutenberg#66827]
**Description**

There's some fragmentation in the behavior of the inserter. When I click on the inserter in the context of the content, sometimes it focuses on the search bar. It doesn't if the page is new and there's no content. If I open the block inserter at the top left side of the screen (opening list view), then it doesn't focus on the search bar.

It would be nice to automatically focus the search bar wherever the block inserter is opened. This way, we improve the cohesion and overall experience, it becomes more usable, and in some places, we reduce the number of clicks users need to find what they're looking for.

Use case: When working and testing TT5, I wanted to insert patterns. Every single time I opened the inserter from the sidebar, I had to look for the search bar and click there before starting my search. Website creators will likely know the block name and/or pattern they want to use. Most users will likely also have one hand on the keyboard and the other on the mouse/trackpad. I would have had a way smoother experience if I didn't have to click the search box every time I was trying to insert something.

https://github.com/user-attachments/assets/a082810e-c3e1-4277-b47b-8c40276e9b35

**Comment by Mamaduka** [comment]
I think autofocus on search from global inserter was removed in recent refactoring. I don't recall all the details, but @jeryj or @afercia might.

**Comment by jeryj** [comment]
Yes, it was intentionally removed. If I recall, it was due to the search being moved beneath the tabs. If we auto-focus the search, then we have skipped valid tabstops and screen reader users won't be alerted that they have other tabs above the search input. If we want to focus the search input like before, we also need to have the search field be the first element within the sidebar.

**Comment by afercia** [comment]
> Yes, it was intentionally removed. If I recall, it was due to the search being moved beneath the tabs. If we auto-focus the search, then we have skipped valid tabstops

Yes. However, the relocation of the search isn't ideal to me. I would agree the search should receive initial focus but then it should be placed above the tabs and be the first thing in the DOM order within the inserter panel. In any case, managing focus should not skip important parts of the UI.

It is worth mentioning that initial focus on the various panels is a broader issue and it's largely inconsistent. For example, when opening the List View and there are blocks, the initial focus goes on the first item thus skipping the tabs.

Overall, I'd suggest to reconsider initial focus on all panels holistically and, more importantly, implement it in a consistent way for all panels.

**Comment by juanfra** [comment]
Thank you for providing the context! I wasn’t aware of the previous discussion. I find it a bit confusing that the search bar isn’t at the top level, especially since it searches for both blocks and patterns. Perhaps it’s just muscle memory—I kept clicking on the list view, expecting to search for blocks or patterns automatically.

I’ve read through the issue and the related PRs, and I understand the reasoning behind the change. That said, I have to admit that this has been a pain point for me when using the editor to create, and I believe this could affect the overall experience of creators.

**Comment by afercia** [comment]
> Thank you for providing the context! I wasn’t aware of the previous discussion. I find it a bit confusing that the search bar isn’t at the top level, especially since it searches for both blocks and patterns

I'd totally agree. To me, the search should be the first thing at the top. For what is worth, i disagree with the design that moved it after the tabs. I'd appreciate some feedback to reconsider that design choice Cc @WordPress/gutenberg-design

**Comment by jameskoster** [comment]
I agree that placing the input beneath the tabs suggests it will search _within the active tab_.

However its worth noting search isn't global—media files aren't included. 

So without re-architecting the Inserter I'm not sure there's a clear change we can make to improve things?

**Comment by Brugman** [comment]
Yes please. A shortcut for "Toggle block inserter" would speed use up even more.
